core/src/models/Destination.ts

```diff
@@ -484,7 +484,10 @@ export class Destination extends LoggedModel<Destination> {
   @BeforeDestroy
   static async waitForPendingExports(instance: Destination, { transaction }) {
     const pendingExportCount = await instance.$count("exports", {
```

Comment:
It seemed important to be able to delete a destination if it ever had an error, but I still tried to keep the same semantics. Still some possible race conditions that would create errors in Resque.

Reply:
There's forthcoming work to change how Destinations get deleted in general - to be more like Groups and they can "drain" and delete themselves fully